app/src/main/java/com/example/recipeopedia/activities/ViewAttemptsActivity.java

```diff
@@ -0,0 +1,123 @@
+        attemptAdapter = new AttemptAdapter(mAttempts);
+        rvAttempts = findViewById(R.id.rvAttempts);
+        rvAttempts.setAdapter(attemptAdapter);
+        rvAttempts.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+        queryAttempts();
+
+        favoriteRecipe = Parcels.unwrap(getIntent().getParcelableExtra(FavoriteRecipe.class.getSimpleName()));
+        etYourAttempt = findViewById(R.id.etYourAttempt);
+        btnPostAttempt = findViewById(R.id.btnPostAttempt);
+        ParseUser currentUser = ParseUser.getCurrentUser();
+        btnPostAttempt.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                favoriteRecipe = Parcels.unwrap(getIntent().getParcelableExtra(FavoriteRecipe.class.getSimpleName()));
+                String description = etYourAttempt.getText().toString();
+                if (description.isEmpty()) {
+                    Toast.makeText(getApplicationContext(), "Description cannot be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                else {
+                    try {
+                        saveAttempt(favoriteRecipe.getRecipeName(), currentUser, description);
+                        attemptAdapter.clear();
+                        queryAttempts();
+                        swipeContainer.setRefreshing(false);
+                        Toast.makeText(getApplicationContext(), "Attempt posted!", Toast.LENGTH_SHORT).show();
+                    } catch (ParseException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        });
+    }
+
+    private void saveAttempt(String recipeName, ParseUser currentUser, String description) throws ParseException {
+        Attempt attempt = new Attempt();
+        attempt.setRecipeName(recipeName);
+        attempt.setUser(currentUser);
+        attempt.setAttempt(description);
+        attempt.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    return;
+                }
+            }
+        });
+    }
+
+    private void queryAttempts() {
+        ParseQuery<Attempt> query = ParseQuery.getQuery(Attempt.class);
+        query.include(Review.KEY_USER);
+        favoriteRecipe = Parcels.unwrap(getIntent().getParcelableExtra(FavoriteRecipe.class.getSimpleName()));
+        String recipeName = favoriteRecipe.getRecipeName();
+        query.whereEqualTo(Attempt.KEY_RECIPE_NAME, recipeName);
+        query.setLimit(20);
+        query.addDescendingOrder("createdAt");
+        query.findInBackground(new FindCallback<Attempt>() {
+            @Override
+            public void done(List<Attempt> attempts, ParseException e) {
+                if (e != null) {
+                    return;
+                }
+                mAttempts.addAll(attempts);
```

Comment:
does this definitely work? `mAttempts` lives in the activity, but the recyclerview is displaying the list that lives in the adapter. personally i wouldn't have an `mAttempts` field at all, I would just have the default constructor for the adapter initialize an empty list and access that through the adapter. either by making the list public or providing getter/setter methods